src/v/cluster/jumbo_log_stm.cc

```diff
@@ -106,6 +106,13 @@ ss::future<jumbo_log::write_intent_id_t> jumbo_log_stm::create_write_intent(
     batch_builder.add_raw_kv(
       serde::to_iobuf(create_write_intent_cmd::key), serde::to_iobuf(record));
 
+    // TODO(nv): Do we want to validate that the object exists in s3?
+    // If we don't a bug in the uploader returns success but the object is not
+    // in s3, we will have a dangling write intent which we won't be able to
+    // clean up. An alternative failure scenario is the uploader uploading the
+    // object at the wrong path. How risky is this? A head request to s3 would
+    // be a good way to validate the object exists.
```

Comment:
>     // TODO(nv): Do we want to validate that the object exists in s3?

maybe, but i wouldn't put it here. validation of the object existing in s3 sounds more like it belong to a scrubbing process.

> dangling write intent which we won't be able to clean up

why would it not be able to be cleaned up? if the problem is that we don't want to depend on timeouts (i agree, that isn't ideal) then we can probably solve this problem with another form of fencing.

Reply:
> why would it not be able to be cleaned up? if the problem is that we don't want to depend on timeouts (i agree, that isn't ideal) then we can probably solve this problem with another form of fencing.

The concern is that batcher might create bad files/not create them at all due to a bug, then write intent is created, then write intent is replicated to user NTPs and offsets are allocated.

What can we do here if jumbo log can't read the data files from s3? Validating that file exists in s3 doesn't give much guarantees though. The file can be corrupted/unreadable anyway. Or, deleted by some other odd process.